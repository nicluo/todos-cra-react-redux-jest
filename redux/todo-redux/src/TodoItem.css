.Checkbox {
    width: 30px;
    height: 30px;
    margin-right: 30px;
    background-color: white;
    border-radius: 50%;
    border: 1px solid #ddd;
    -webkit-appearance: none;
    outline: none;
    cursor: pointer;
}

.Checkbox:checked {
    background-image: url('Check.svg');
    background-size: 30px;
}
