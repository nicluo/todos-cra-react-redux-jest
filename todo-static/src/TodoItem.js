import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TodoItem extends Component {
  handleChange = (event) => {
    this.props.onChange(this.props.id, event.target.checked);
  };

  render() {
    const {completed, children} = this.props;

    return (
      <div className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3">
        <div className="flex items-center pa3 f4 f4-ns lh-copy">
          <input type="checkbox" className="Checkbox" checked={completed} onChange={this.handleChange} />
          <div className="dib">{children}</div>
        </div>
      </div>
    );
  }
}

TodoItem.propTypes = {
  id: PropTypes.number.isRequired,
  completed: PropTypes.bool.isRequired,
  children: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default TodoItem;
