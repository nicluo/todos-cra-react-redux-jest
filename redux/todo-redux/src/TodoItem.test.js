import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import TodoItem from './TodoItem';

it('Renders empty children without crashing', () => {
  const noop = () => {};

  const component = shallow(
    <TodoItem id={1} completed={false} onChange={noop}></TodoItem>
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});

it('Displays item content passed as child string', () => {
  const noop = () => {};

  const component = shallow(
    <TodoItem id={1} completed={false} onChange={noop}>Task 1</TodoItem>
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});

it('Displays checked status when passed completed boolean', () => {
  const noop = () => {};

  const component = shallow(
    <TodoItem id={1} completed={false} onChange={noop}>Task 1</TodoItem>
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});

it('Calls onChange when checkbox is clicked', () => {
  const onChange = sinon.spy();
  const component = shallow(
    <TodoItem id={1} completed={false} onChange={onChange}>Task 1</TodoItem>
  );
  component.find('input[type="checkbox"]').simulate('change', { target: { checked: true } });
  expect(onChange.calledOnce).toEqual(true);
});

it('Calls onChange when checkbox is clicked when already completed', () => {
  const onChange = sinon.spy();
  const component = shallow(
    <TodoItem id={1} completed={true} onChange={onChange}>Task 1</TodoItem>
  );
  component.find('input[type="checkbox"]').simulate('change', { target: { checked: false } });
  expect(onChange.calledOnce).toEqual(true);
});
