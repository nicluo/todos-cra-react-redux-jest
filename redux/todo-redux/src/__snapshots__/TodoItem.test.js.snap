// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays checked status when passed completed boolean 1`] = `
<div
  className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3"
>
  <div
    className="flex items-center pa3 f4 f4-ns lh-copy"
  >
    <input
      checked={false}
      className="Checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="dib"
    >
      Task 1
    </div>
  </div>
</div>
`;

exports[`Displays item content passed as child string 1`] = `
<div
  className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3"
>
  <div
    className="flex items-center pa3 f4 f4-ns lh-copy"
  >
    <input
      checked={false}
      className="Checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="dib"
    >
      Task 1
    </div>
  </div>
</div>
`;

exports[`Renders empty children without crashing 1`] = `
<div
  className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3"
>
  <div
    className="flex items-center pa3 f4 f4-ns lh-copy"
  >
    <input
      checked={false}
      className="Checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="dib"
    />
  </div>
</div>
`;
